cmd/billing/billing.go

```diff
@@ -0,0 +1,197 @@
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		clusterData := respC.Data.GetClusters()
+		var clusterUUIDMap = make(map[string]string)
+
+		for _, cluster := range clusterData {
+			clusterName := cluster.Name
+			clusterUUIDMap[clusterName] = cluster.Id
+		}
+
+		var selectedUUIDs []string
+		if len(clusters) > 0 {
+			selectedUUIDs = make([]string, 0, len(clusters))
+			for _, clusterName := range clusters {
+				if clusterID, ok := clusterUUIDMap[clusterName]; ok {
+					selectedUUIDs = append(selectedUUIDs, clusterID)
+				} else {
+					logrus.Warnf("Cluster name '%s' not found in clusterUUIDMap", clusterName)
+				}
+			}
+		} else {
+			selectedUUIDs = nil
+		}
+
+		resp, r, err := authApi.GetBillingUsage(startDate, endDate, selectedUUIDs).Execute()
+
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		usageData := resp.GetData()
+
+		switch strings.ToLower(format) {
+		case "csv":
+			if err := outputCSV(usageData, "usage.csv"); err != nil {
+				return
+			}
+		case "json":
+			if err := outputJSON(usageData, "usage.json"); err != nil {
+				return
+			}
+		default:
+			logrus.Warnf("Unsupported format: %s. Defaulting to CSV.", format)
+			if err := outputCSV(usageData, "usage.csv"); err != nil {
+				return
+			}
+		}
+	},
+}
+
+func getDimensionNames(resp ybmclient.BillingUsageData) []string {
+	var dimensionNames []string
+
+	if len(resp.GetDimensions()) > 0 {
+		for _, dimension := range resp.GetDimensions() {
+			dimensionNames = append(dimensionNames, string(dimension.Name))
+		}
+	}
+
+	return dimensionNames
+}
+
+func outputCSV(resp ybmclient.BillingUsageData, filename string) error {
+	file, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	writer := csv.NewWriter(file)
+	defer writer.Flush()
+
+	dimensionHeaders := []string{"Date"}
+	for _, dimension := range resp.GetDimensions() {
+		dimensionHeaders = append(dimensionHeaders, string(dimension.GetName())+"_Daily", string(dimension.GetName())+"_Cumulative")
+	}
+
+	err = writer.Write(dimensionHeaders)
+	if err != nil {
+		return err
+	}
+
+	for _, dataPoint := range resp.GetDimensions()[0].DataPoints {
+		row := []string{dataPoint.GetStartDate()}
+
+		for _, dimension := range resp.GetDimensions() {
+			var found bool
+			for _, point := range dimension.DataPoints {
+				if point.StartDate == dataPoint.StartDate {
+					row = append(row, fmt.Sprintf("%f", *point.DailyUsage))
+					row = append(row, fmt.Sprintf("%f", *point.CumulativeUsage))
+					found = true
+					break
+				}
+			}
+
+			// If no data found for the current dimension, add placeholders
+			if !found {
+				row = append(row, "0.000000", "0.000000")
+			}
+		}
+
+		err := writer.Write(row)
+		if err != nil {
+			return err
+		}
+	}
+
+	fmt.Printf("CSV data written to %s\n", filename)
```

Comment:
we can add some color as we do in others command

Reply:
Done